src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Emitter/Helpers.cs

```diff
@@ -229,18 +229,30 @@ private void EmitBlankLineIfRequired()
                 _emitBlankLineBeforeNextStatement = true;
             }
 
-            private void EmitCheckForNullArgument_WithBlankLine(string paramName, bool voidReturn = false)
+            private void EmitCheckForNullArgument_WithBlankLine(string paramName, bool useThrowIfNullMethod, bool voidReturn = false)
             {
-                string returnExpr = voidReturn
-                    ? "return"
-                    : $"throw new ArgumentNullException(nameof({paramName}))";
-
-                _writer.WriteLine($$"""
+                if (voidReturn)
+                {
+                    _writer.WriteLine($$"""
                     if ({{paramName}} is null)
                     {
-                        {{returnExpr}};
+                        return;
                     }
                     """);
+                }
+                else
+                {
+                    string throwIfNullExpr = useThrowIfNullMethod
+                    ? $"ArgumentNullException.ThrowIfNull({paramName});"
```

Comment:
>ArgumentNullException [](http://example.com/codeflow?start=24&length=21)

do we need full qualified names here? like global::System.ArgumentNullException?

Reply:
I see we are emitting `using System` so maybe we don't need the full qualified name.

Reply:
Exactly, a recent change deleted the need for global imports like this